File: src/test/java/Pages/PageLogin.java
package Pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

public class PageLogin {

    WebDriver driver;


    By username = By.id("txtUsername");
    By pass = By.id("txtPassword");
    By login = By.id("btnLogin");
    By perfil = By.id("welcome");
    By logout = By.linkText("Logout");

    public PageLogin(WebDriver driver){

        this.driver = driver;
    }


    public void setUsername(){
        driver.findElement(username).sendKeys("Admin");
    }

    public void setPass(){
        driver.findElement(pass).sendKeys("admin123");
    }

    public void setLogin(){
        driver.findElement(login).click();
    }
    public void setPerfil(){
        driver.findElement(perfil).click();
    }
    public void setLogout(){
        driver.findElement(logout).click();
    }
}

File: src/test/Kotlin/Pages/PageLogin.kt
package Pages


import org.openqa.selenium.support.FindBy
import org.openqa.selenium.WebDriver
import org.openqa.selenium.WebElement
import org.openqa.selenium.support.PageFactory

class PageLogin(var driver: WebDriver) {

    init {
        PageFactory.initElements(driver, this)
    }

    @FindBy(id = "txtUsername")
    var username: WebElement? = null
    @FindBy(id = "txtPassword")
    var pass: WebElement? = null

    @FindBy(id = "btnLogin")
    var login: WebElement? = null

    @FindBy(id = "welcome")
    var profile: WebElement? = null

    @FindBy(linkText = "Logout")
    var logout: WebElement? = null


    fun setUsername() {
        username!!.sendKeys("Admin")
    }

    fun setPass() {
        pass!!.sendKeys("admin123")
    }

    fun setLogin() {
        login!!.click()
    }

    fun setProfile() {
        profile!!.click()
    }

    fun setLogout() {
        logout!!.click()
    }
}